modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/MessageBoardMessageResourceImpl.java

```diff
@@ -404,6 +430,23 @@ private MessageBoardMessage _addMessageBoardMessage(
 			contextAcceptLanguage.getPreferredLocale());
 	}
 
+	private MessageBoardMessage _getMessageBoardMessageByExternalReferenceCode(
```

Comment:
👍

Reply:
At first I wanted to add a find method at Service level but as they were so many problems touching the services, I've decided to keep it in a method here

Reply:
Yeah, best leave it here, the exception belongs to us more than in the service level.